### DISCUSSION TITLE
How to use middleware to match trailing slash in redirects

### DISCUSSION THREAD

**Discussion by shareoc** [vercel/community#1027] [Help]
Hi!

I'm new to Vercel. I want to achieve something similar to what Netlify does with [URL Normalization](https://docs.netlify.com/routing/redirects/redirect-options/#trailing-slash). The redirect rules in Netlify will match paths to rules regardless of whether or not they contain a trailing slash. For instance, take the following redirect rule:

```
    {
      "source": "/blog-categories/",
      "destination": "/blog/categories/",
      "permanent": true
    }
```

This redirect rule should redirect both `/blog-categories/` and `/blog-categories` to `/blog/categories/` without having to define two redirect rules.

Using [Edge Middleware](https://vercel.com/docs/concepts/functions/edge-middleware) I was able to append a trailing slash to all requests without a trailing slash, basically reproducing what [`trailingSlash`](https://vercel.com/docs/project-configuration#project-configuration/trailing-slash) already does. However, this creates a redirect chain, as `/blog-categories` is redirected first to `/blog-categories/`, which is subsequently forwarded to `/blog/categories/`. 

What would be the right approach to match redirect paths regardless of the trailing slash while avoiding redirect chains?

Thank you in advance for any help!

Edit: I'm using **Gatsby**

**Comment by acgourley** [comment]
Did you ever sort this out?
